scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -942,14 +952,21 @@ class FormatOps(
 
   def typeTemplateSplits(
       template: Template,
-      indent: Int
-  ): Seq[Split] = {
+      indent: Int,
+      isDerives: Boolean = false
+  )(implicit line: sourcecode.Line): Seq[Split] = {
     val hasSelfAnnotation = template.self.tokens.nonEmpty
-    val expire = (template.parent match {
+    val curlyOrDerivesOrWith = template.parent match {
       case Some(_: Defn.Given) =>
         findLast(template.tokens)(x => x.is[T.KwWith] && owners(x) == template)
-      case _ => templateCurly(template)
-    }).getOrElse(template.tokens.last)
+      case _ if isDerives => templateCurly(template)
+      case _ => templateDerivesOrCurly(template)
+    }
+
+    val (expire, when) = curlyOrDerivesOrWith match {
+      case Some(value) => (value, ExpiresOn.Before)
```

Comment:
why do we expire before the token which comes before `derives`? and not before `derives`, or after that preceding token?

Reply:
Hmm... honestly, I am not sure. It didn't work any other way. I will dig in and see what's going on.

Reply:
Ok, I found the issue, when moving to the previous token we would stop at a Space token, we need to get back to a non whitespace one, which is now changed.